docs/DeveloperGuide.adoc

```diff
@@ -170,168 +216,116 @@ Classes used by multiple components are in the `tatracker.commons` package.
-** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedTaTracker`.
-* **Alternative 2:** Use `HistoryManager` for undo/redo
-** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
-** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
-// end::undoredo[]
+[[Student-View]]
+=== Student View
+*Student view* is the term used to characterise the different functionalities
+related to the students.
 
-//tag::moduleview[]
-=== Module View
-Module view is the term used to characterise the different functionalities
-related to the modules and groups that the user is affiliated with.
+Students are a part of *groups* and groups are a part of *modules*.
 
 ==== Model Framework
 The following class diagram shows how different classes are related in the
-functioning of the module view.
+functioning of the Student View.
 
 .Module View - Class Diagram
 image::ModuleModelClassDiagram.png[]
 
-The TaTracker model class contains a UniqueModuleList which helps it keep track
-of the different modules the user is associated with. Each module contains a
-UniqueGroupList and a UniqueSessionList.
+The TaTracker `Model` class contains a `UniqueModuleList` which helps it keep track
+of the different *modules* the user is associated with. Each `module` contains a
+`UniqueGroupList` and a `UniqueSessionList`.
+
+The `UniqueGroupList` contains a list of all the *groups* of a module that the user
+is affiliated with. Each group contains a `UniqueStudentsList` that contains the
+*students* in that group.
 
-The UniqueGroupList contains a list of all the groups of a module that the user
-is affiliated with. Each group contains a UniqueStudentsList that contains the
-students in that group.
+The `UniqueSessionList` contains a list of all the *sessions that have already been
+done* associated with the module. This list is used in the TSS view.
 
-The UniqueSessionList contains a list of all the done sessions associated with the module.
-This list is used in the TSS view.
+====
+*Alternative Implementation*
 
-==== Implementation of the Module Add and Delete Commands
+* An alternative implementation would be to a `UniqueModuleList` for all modules, a `list`
+for all groups and a `list` for all students and then filter by module code or group
+code according to how the user would want it to be.
+
+* While this would be easier to
+implement, it would also introduce cyclic dependencies as not only would students
+need to keep a track of their groups, but groups would also keep a track of their students.
```

Comment:
Not really. This can be fixed with association classes, although it will require more commands.

Here's an alternative:
The module can have a list of enrolled students and a list of registered modules.
And a group can have a list of enrolled students.

Reply:
Reason not doing this is cause at the moment it does not make sense to have students outside a module or group.

Reply:
Okie, will mention this.